crates/ruff_python_parser/src/lexer.rs

```diff
@@ -429,6 +429,31 @@ impl<'source> Lexer<'source> {
                     value.push('\\');
                 }
                 '\n' | '\r' | EOF_CHAR => {
```

Comment:
Would it make sense to have a branch for `?` to avoid the reverse inspection of the just lexed string?

```
			`'?' => {
				let mut question_count = 1u32;
				while self.cursor.eat_char('?') {
					question_count += 1;
				}

				if self.cursor.first == '\n' | '\r' | EOF_CHAR {
					// Decide whether to include the `?` 

					// magic help
				} else {
					// Never mind
					value.reserve(question_count);
					for 0..question_count {
						value.push('?');
					}
				}
			}
```

Reply:
My initial implementation used a branch method but the way I was doing it required some state variables, etc. This implementation looks actually reasonable. Let me check, I think this could be doable.